drivers/adc/adc_ads114s0x.c

```diff
@@ -1080,7 +1080,7 @@ static int ads114s0x_gpio_write_config(const struct device *dev)
 {
 	struct ads114s0x_data *data = dev->data;
 	const struct ads114s0x_config *config = dev->config;
-	uint8_t register_addresses[2];
+	enum ads114s0x_register register_addresses[2];
```

Comment:
This commit and the following commits are not related to the GPIO shell changes.  Please move to a separate PR.

Reply:
This issue was was one of many with the ads114s0x_gpio driver/test that were causing CI to fail for this PR. I've opened https://github.com/zephyrproject-rtos/zephyr/pull/64876

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/64876 has merged.  Please rebase this PR.

Reply:
Done 👍